File: PeptideHitResultsProcessor/clsSearchResultsMSGFDB.vb
Option Strict On

' This class is used to track the peptide details for an MSGF+ search result
' See clsSearchResultsBaseClass for additional information
'
' -------------------------------------------------------------------------------
' Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
' Created 8/12/2011
'
' E-mail: matthew.monroe@pnnl.gov or matt@alchemistmatt.com
' Website: http://omics.pnl.gov/ or http://www.sysbio.org/resources/staff/
' -------------------------------------------------------------------------------

Imports PHRPReader

Public Class clsSearchResultsMSGFDB
    Inherits clsSearchResultsBaseClass

#Region "Classwide Variables"
    ' Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables 
#End Region

#Region "Properties"

    Public Property FragMethod As String
    Public Property NTT As String

    Public Property DeNovoScore As String
    Public Property MSGFScore As String
    Public Property SpecEValue As String        ' SpecProb in MSGFDB; SpecEValue in MSGF+
    Public Property RankSpecEValue As String
    Public Property EValue As String            ' PValue in MSGFDB; EValue in MSGF+

    Public Property QValue As String            ' Will contain target/decoy FDR when -tda 1 was used; will contain EFDR when -tda 1 was not used; FDR in MSGFDB; QValue in MSGF+
    Public Property PepQValue As String         ' Only present if searched using -tda 1; PepFDR in MSGFDB; PepQValue in MSGF+

    Public Property PrecursorMZ As String
    Public Property MSGFPlusComputedDelM As String
    Public Property MSGFPlusComputedDelMPPM As String

    Public Property IsotopeError As String      ' Only reported by MSGF+

    Public Property MSGFPlusResults As Boolean
#End Region

    Public Sub New(objPeptideMods As clsPeptideModificationContainer, peptideSeqMassCalculator As clsPeptideMassCalculator)
        ' Note that the following call will call both the base class's Clear sub and this class's Clear Sub
        MyBase.New(objPeptideMods, peptideSeqMassCalculator)
    End Sub

    Public Overrides Sub Clear()
        MyBase.Clear()

        FragMethod = String.Empty
        NTT = String.Empty

        DeNovoScore = String.Empty
        MSGFScore = String.Empty
        SpecEValue = String.Empty
        RankSpecEValue = String.Empty
        EValue = String.Empty

        QValue = String.Empty
        PepQValue = String.Empty

        PrecursorMZ = String.Empty
        MSGFPlusComputedDelM = String.Empty
        MSGFPlusComputedDelMPPM = String.Empty

        IsotopeError = String.Empty
        MSGFPlusResults = False
    End Sub

End Class

File: PeptideHitResultsProcessor/clsSearchResultsMSGFDB.cs
// -------------------------------------------------------------------------------
// Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
//
// E-mail: matthew.monroe@pnnl.gov or proteomics@pnnl.gov
// Website: https://omics.pnl.gov/ or https://www.pnnl.gov/sysbio/ or https://panomics.pnnl.gov/
// -------------------------------------------------------------------------------
//
// Licensed under the 2-Clause BSD License; you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
// https://opensource.org/licenses/BSD-2-Clause
//
// Copyright 2018 Battelle Memorial Institute

using PHRPReader;

namespace PeptideHitResultsProcessor
{
    /// <summary>
    /// This class is used to track the peptide details for an MSGF+ search result
    /// See clsSearchResultsBaseClass for additional information
    /// </summary>
    public class clsSearchResultsMSGFDB : clsSearchResultsBaseClass
    {
        #region "Classwide Variables"
        // Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables
        #endregion

        #region "Properties"

        public string FragMethod { get; set; }
        public string NTT { get; set; }

        public string DeNovoScore { get; set; }
        public string MSGFScore { get; set; }
        public string SpecEValue { get; set; }      // SpecProb in MSGFDB; SpecEValue in MSGF+
        public string RankSpecEValue { get; set; }
        public string EValue { get; set; }          // PValue in MSGFDB; EValue in MSGF+

        // ReSharper disable once CommentTypo
        public string QValue { get; set; }          // Will contain target/decoy FDR when -tda 1 was used; will contain EFDR when -tda 1 was not used; FDR in MSGFDB; QValue in MSGF+
        public string PepQValue { get; set; }       // Only present if searched using -tda 1; PepFDR in MSGFDB; PepQValue in MSGF+

        public string PrecursorMZ { get; set; }
        public string MSGFPlusComputedDelM { get; set; }
        public string MSGFPlusComputedDelMPPM { get; set; }

        public string IsotopeError { get; set; }    // Only reported by MSGF+

        public bool MSGFPlusResults { get; set; }
        #endregion

        // Note that the following call will call both the base class's Clear method and this class's Clear method
        public clsSearchResultsMSGFDB(clsPeptideModificationContainer peptideMods, clsPeptideMassCalculator peptideSeqMassCalculator)
            : base(peptideMods, peptideSeqMassCalculator)
        {
        }

        public override void Clear()
        {
            base.Clear();

            FragMethod = string.Empty;
            NTT = string.Empty;

            DeNovoScore = string.Empty;
            MSGFScore = string.Empty;
            SpecEValue = string.Empty;
            RankSpecEValue = string.Empty;
            EValue = string.Empty;

            QValue = string.Empty;
            PepQValue = string.Empty;

            PrecursorMZ = string.Empty;
            MSGFPlusComputedDelM = string.Empty;
            MSGFPlusComputedDelMPPM = string.Empty;

            IsotopeError = string.Empty;
            MSGFPlusResults = false;
        }
    }
}
